pkg/devfile/parser/parse_test.go

```diff
@@ -4230,7 +4227,7 @@ func Test_parseFromURI_PrivateToken(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := parseFromURI(tt.importReference, tt.curDevfileCtx, &resolutionContextTree{}, resolverTools{token: tt.token})
+			got, err := parseFromURI(tt.importReference, tt.curDevfileCtx, &resolutionContextTree{}, resolverTools{})
```

Comment:
The purpose of this test is not to validate if the token got set since this is already known and it is part of your setup.  What should be tested are your 3 supported scenarios:

1.  Verify that you can parse the devfile contents for a private main devfile URL
2.  Verify that you can parse the devfile contents for your private main devfile with a private parent reference
3.  Verify you can can parse the devfile contents for your private main devfile with public parent reference 

Those are the happy path scenarios.  There should also be negative path testing which can include: 

1. Parsing a private main devfile without a valid token
2. Parsing a bad devfile URL (malformed but vendors are valid) with/without a valid token - does this fail gracefully?
3.  Parsing a bad devfile URL (unsupported vendors), etc

These are just some examples that you can start with.  

This means you will need to set up your test servers to handle requests with tokens.  You can take a look at this [mock server](https://github.com/redhat-appstudio/application-service/blob/main/pkg/github/mock.go#L46) for some ideas.